arch/i386/mcount.S

```diff
@@ -0,0 +1,50 @@
+/* in i386, generally used stack for argument passing. */
+/* use register for return : %eax */
+/* no need save registers */
+/* stack frame (with -pg): parent addr = 4(%ebp) */
+/* child addr = (%esp) */
+
+#include "utils/asm.h"
+
+GLOBAL(mcount)
+	sub $32, %esp
+	/* save registers */
+	movl %edx, 24(%esp)
+	movl %ecx, 20(%esp)	
```

Comment:
I suggest save `%eax` as well just for sure.  I saw some environment passed a hidden argument via `%rax` on x86_64.